### DISCUSSION TITLE
Exclude detection of certain languages or detect only one particular language

### DISCUSSION THREAD

**Discussion by sreena7h** [openai/whisper#1549] [General]
I specifically needs to detect english language only and don't want to detect any native languages in the Whisper API.

Is there any option for that?

**Comment by ryanheise** [comment]
Unfortunately, I'm not sure if I understand what it means to detect only English. If you just want to classify between "English" and "Not English", then you can just run Whisper and get back the result. If it's anything other than English, you can treat it as "Not English". There is no additional expense in doing that. Otherwise, maybe you don't mean the same thing by "detection" as is commonly understood. Although since 2 other people upvoted your question immediately, I guess 2 other people have a common understanding to you :-) In any case, though, it will help if you explain further what you personally mean by this, as I unfortunately don't understand it.

**Reply by sreena7h** [reply]
I tried with an audio in my native language **malayalam** and it detected some random English sentence. 

Note:
In the API request, I was provided with `model='whisper-1'` and `language='en'`

**Reply by ryanheise** [reply]
But I still don't understand the meaning as explained in my previous reply. Maybe it would help if you answer this: What do you want Whisper to do in the situation where you give it audio in Malayalam?

**Reply by sreena7h** [reply]
Since I specified language='en' it should not detect any malayalam words

**Reply by ryanheise** [reply]
I see, I understand at least part of what you are trying to say now, but not all.

When you specify language='en', you are actually disabling detection and telling it to assume English rather than try to detect which language it actually is. So in your scenario, there is no actual detection taking place, but rather, assuming.

You need to omit the language option if you want detection to take place. Then you can look at the result and see if it detected English or Not English.
